### PR TITLE
Fix MaxMessageLength off-by-one in Protobuf payload decoder (#4436)

### LINKED ISSUE #4436: [Audit-Low] MaxMessageLength is enforced off by one
> **AI-generated audit finding** — this issue was opened from an automated security/correctness audit. It **has not been triaged by a human yet**; verify the reasoning, reproducibility, and severity before acting on it.

### Low: `MaxMessageLength` is enforced off by one — **CONFIRMED**

Affected code:

- [src/IceRpc.Protobuf/RpcMethods/Internal/PipeReaderExtensions.cs:116-119](https://github.com/icerpc/icerpc-csharp/blob/d6354eca1dc6df4b58f2115b01848c7b66a7bed4/src/IceRpc.Protobuf/RpcMethods/Internal/PipeReaderExtensions.cs#L116) — `if (messageLength >= maxMessageLength)`
- [src/IceRpc.Protobuf/Features/ProtobufFeature.cs:13](https://github.com/icerpc/icerpc-csharp/blob/d6354eca1dc6df4b58f2115b01848c7b66a7bed4/src/IceRpc.Protobuf/Features/ProtobufFeature.cs#L13), [src/IceRpc.Protobuf/Features/IProtobufFeature.cs:11](https://github.com/icerpc/icerpc-csharp/blob/d6354eca1dc6df4b58f2115b01848c7b66a7bed4/src/IceRpc.Protobuf/Features/IProtobufFeature.cs#L11) — contract describes the value as the maximum permitted length

Verification:

Confirmed. `>=` rejects a message whose body length equals `maxMessageLength`. `MaxMessageLength = 1024` refuses a 1024-byte payload; `MaxMessageLength = 0` refuses *every* payload including empty ones. Both behaviors contradict the property name and docs.

Impact:

- Boundary-sized valid messages are rejected unexpectedly.
- `MaxMessageLength = 0` is silently unsatisfiable.

Recommendation:

- Change the check to `messageLength > maxMessageLength`.
- Add boundary tests for exact-limit and zero-length messages.

Status: **Valid, Low severity**.

---

**Source report:** [src-IceRpc.Protobuf-audit-2026-04-14.md](https://github.com/icerpc/icerpc-csharp-audit/blob/main/src-IceRpc.Protobuf-audit-2026-04-14.md) (finding ``MaxMessageLength` is enforced off by one — **CONFIRMED**`)

**Severity (auditor-assigned):** Low

### PR DESCRIPTION
**Pull request by pepone**
Fix #4436

### CODE REVIEW

**Review by externl — APPROVED** [review]

File: tests/IceRpc.Slice.Tests/PipeReaderTests.cs [hunk 1]
```diff
@@ -112,4 +112,51 @@ public void Trying_to_read_a_short_segment_fails()
             () => _ = pipeReader.TryReadSliceSegment(maxSize: 100, out ReadResult readResult),
             Throws.TypeOf<InvalidDataException>());
     }
+
+    [Test]
+    public async Task Reading_a_segment_with_size_equal_to_max_size_succeeds()
+    {
+        // 20 = 4 * 5 means the payload size is 5
```
Comment (InsertCreativityHere) [inline tests/IceRpc.Slice.Tests/PipeReaderTests.cs]: I think `20 = 5 << 2` would be more true to what's going on here, instead of just saying it's multiplied by 4.

**Review by InsertCreativityHere — APPROVED** [review]
Looks good to me!

Also:
I don't think we need a test which ensures `maxValue = 0` works as expected.
